In [27]:
pd="/mnt/solexa/mel_yang/data/2.2M/"
fh="UPA_all8"
p1,p2=("Longlin","Longlin_com")
p1,p2=("Longlin","LiangDaoChineseNeolithic")
p1,p2=("I0879","I0906") #LD
p1,p2=("L2012","L2039") #Daxi
p1,p2=("L0972.3lib","L2008") #XY
p1ind,p2ind=([],[])
bypop=False
indfile=open(pd+fh+".ind",'r')
for ind,line in enumerate(indfile):
    x=line.split()
    if bypop==True:
        if x[-1] == p1: p1ind.append(ind)
        elif x[-1] == p2: p2ind.append(ind)
        else: continue
    if bypop==False:
        if x[0] == p1: p1ind.append(ind)
        elif x[0] == p2: p2ind.append(ind)
        else: continue
indfile.close()

combos=["00","01","02","09","11","12","19","22","29","99"]
totalvals=[1,1,1,0,1,1,0,1,0,0]
countvals=[0,0.5,1,0,0.5,0.5,0,0,0,0]
diffvals={i:(totalvals[ind],countvals[ind]) for ind,i in enumerate(combos)}
#print diffvals

mynames=[("%s_%i" % (p1,ind1), "%s_%i" % (p2,ind2)) for ind1,i in enumerate(p1ind) for ind2,i in enumerate(p2ind)]
totals=[0.0 for i in mynames]
counts=[0.0 for i in mynames]

genofile=open(pd+fh+".geno",'r')
for pos,line in enumerate(genofile):
    p1genos=[line[i] for i in p1ind]
    p2genos=[line[i] for i in p2ind]
    num=0
    for p1geno in p1genos:
        for p2geno in p2genos:
            myset=''.join(sorted(p1geno+p2geno))
            totals[num]+=diffvals[myset][0]
            counts[num]+=diffvals[myset][1]
            num+=1
            #print myset, totals,counts
    #if pos==100: break
genofile.close()

freqs=[i/totals[ind] for ind,i in enumerate(counts)]
print counts, totals, {i:freqs[ind] for ind,i in enumerate(mynames)}

[130706.0] [552958.0] {('L0972.3lib_0', 'L2008_0'): 0.23637599962384123}


### Results from Above

[6647.0] [140850.0] {('Longlin_0', 'Longlin_com_0'): 0.04719204827831026}

[83850.0] [382053.0] {('I0879_0', 'I0906_0'): 0.21947216747414625} #within LiangDao

[9626.0] [43426.0] {('L2012_0', 'L2039_0'): 0.22166444065767052} ##within Daxi

[130706.0] [552958.0] {('L0972.3lib_0', 'L2008_0'): 0.23637599962384123} ##within Xinyi

[17834.0, 32317.0] [74509.0, 132454.0] {('Longlin_0', 'LiangDaoChineseNeolithic_1'): 0.24398659157141347, ('Longlin_0', 'LiangDaoChineseNeolithic_0'): 0.2393536351313264}

Xinyi/Xinyi2 = 723842 SNPs/790502 SNPs
Daxi = 132525/

In [30]:
#Han
print 454771/2144502.0
print 425552/2144502.0
print 428079/2144502.0
print 434832/2144502.0

#Altai
print 105976/2109717.0

0.212063686581
0.198438611855
0.199616974011
0.202765956852
0.050232329739
